### ISSUE TITLE
Is it possible that the GAIB high-confidence VCFs have erroneous sites?

### ISSUE THREAD

**Issue by jumpyknight** [google/deepvariant#317]
Hi.

I tried to investigate the ADs and VAFs of sites with label '0' in the training data.
In most cases, hom-ref sites have very low ADs, which is generated by sequencing errors. However, about 3% of the total examples (about 20000) have AD values greater than 10, yet still labeled as hom-ref. For example, in the BAM I'm studying, the following site

4:39708091-39708091 C->CT DP:AD:VAF 46:28,15:0.326087

is found, and labeled as hom-ref. Most of these sites (about 98% of which) are short-tandem-repeats (STR), the reference bases are 
CTTTTTTTTTTTTTTTTTCCCAGATGGAAT
at the mentioned site.

It seems the GIAB high-confidence VCFs have missed quite a few STR sites (which are within the high-confidence regions).

My question is, would these mislabeled sites pollute the training set of DeepVariant, and lead to a higher FN rate? Could this be fixed by, for example, adding an STR/non-STR channel?

**Comment by MariaNattestad** [comment]
GIAB used a combination of different kinds of sequencing data to get the truth VCF, including long-range technologies that give better results than using short-read sequencing alone.

Learning from this truth set with more context is why DeepVariant performs better than purely looking at AD and VAF alone. DeepVariant learns to balance false positives and false negatives, so when it labels some high-AD loci as '0' it is because they look like sites that GIAB labeled as hom-ref. STRs in particular can cause false positives when using simple allele depth approaches. Since DeepVariant sees the base sequence in the pileup image, it effectively already has an STR channel, so it can "learn" about STRs and take their presence into account.

While the GIAB truth set is not perfect, it has generally performed far better by employing multiple technologies relative to using short-read allele depth alone. I would say it's more likely that these sites were determined by other technologies to be hom-ref than that they were missed by GIAB.
Does that answer your question?

**Comment by jumpyknight** [comment]
Hi @MariaNattestad , thanks for the reply.

I found there's much more STRs inside the GIAB VCFs than the "missed" ones, so I believe GIAB has treated the STRs quite seriously.
However in many sites I found, the repeating units seems to be inserted or deleted as a group, for example, ATATATATAT -> ATATATAT is observed, but ATATATATAT -> ATATTAAT or other cases is not.
Since the allele depths are also high enough, the probability should be very small that such cases are caused by sequencing errors.
I'm not very familiar with sequencing or sample preparation, could other source of errors have more dominate contributions?

On the other hand, STRs can indeed be detected by DeepVariant in the pileup images, but could it help by inputting the information explicitly? I believe that would separate mixed examples in an earlier stage in the data space.

**Comment by AndrewCarroll** [comment]
Hi @jumpyknight 

Which GIAB sample are you using and which version of the truth set? For HG002, there is a new truth set (v4.1) - ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/AshkenazimTrio/analysis/NIST_v4.1_SmallVariantDraftBenchmark_12182019/

Which corrected a number of errors in the prior truth set. If you are using v3.3.2 for this sample, it would be interesting to know if those have been corrected.

To you question about the type of variants that occur. Yes, it is known that repeat expansion and contraction is more likely to occur, and these tend to add to the full repeat content. So it is not unexpected to see changes in repeat number. 

To your question about whether this may have caused DeepVariant to be more prone to call reference, I think the proportion of errors in Genome in a Bottle is quite small. DeepVariant is very accurate, so these may be a reasonable fraction of the total apparent errors, but they are not a large fraction of the total training sites. I don't think that the presence of errors in the training set will have a large impact on DeepVariant, but I it may make DeepVariant just a bit worse than it could otherwise.
